### PR TITLE
Merge python version of Quota Monitoring Solution into PSO repo.

### PR DESCRIPTION
**Pull request by r-kyatham**
Includes the following changes
1. Use Cloud Monitoring data as source for quota information. Java version will be updated to do the
same(right now it only pulls selective compute quota from GCE API's).
2. Move java version to 'java' directory.
3. Pull all quota metrics. 
4. Comprehensive README
5. Terraform automation for deployment

### CODE REVIEW

File: tools/quota-monitoring-alerting/python/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Copy local code to the container image.
+ENV APP_HOME /app
+WORKDIR $APP_HOME
+COPY . ./
+
+# Run the web service on container startup.
+# Use gunicorn webserver with one worker process and 8 threads.
+# For environments with multiple CPU cores, increase the number of workers
+# to be equal to the cores available.
+CMD exec gunicorn --bind :$PORT --workers 1 --threads 8 --timeout 0 main:app
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/Dockerfile:41]: Please use Docker MultiStage Build for this Dockerfile https://docs.docker.com/develop/develop-images/multistage-build/

File: tools/quota-monitoring-alerting/python/bigquery_schemas/dashboard_view.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+WITH
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/bigquery_schemas/dashboard_view.sql:1]: All Source files should have a License Header. Please add one

File: tools/quota-monitoring-alerting/python/config.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Configuration data for building Quota dashboards and pushing custom
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/config.yaml:1]: All Source files should have a License Header. Please add one

File: tools/quota-monitoring-alerting/python/config.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+name: 'quota'
+# Host project settings.
+project: '$PROJECT'
+templates_outputs_dir: 'templates/outputs'
+
+
+# List of metrics and their thresholds for triggering Quota threshold report.
+# fixme: This should accomodate multiple thresholds.
+thresholds:
+          'ALL': 80  # This signifies all have same threshold.
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/config.yaml:12]: Please fix indentation

File: tools/quota-monitoring-alerting/python/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,54 @@
+asgiref==3.3.4
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/requirements.txt:1]: All Source files should have a License Header. Please add one

File: tools/quota-monitoring-alerting/python/main.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+# Below logic is for Cloud Run deployments.
+
+
+@app.route('/message')
+def message():
+    """Hello end point."""
+    return 'Hello\n'
+
+
+@app.route('/project/list')
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/main.py:106]: Is this endpoint supposed to return a list of projects?

File: tools/quota-monitoring-alerting/python/main.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+def projects_list():
+    """Endpoint to recieve project scan request(s)."""
+    return publish_projects(request)
+
+
+@app.route('/project/metric/list', methods=['POST'])
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/main.py:112]: Is this endpoint supposed to return a list of metrics?

File: tools/quota-monitoring-alerting/python/templates/inputs/quota_exceeded_threshold_report_policy.yaml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+combiner: OR
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/templates/inputs/quota_exceeded_threshold_report_policy.yaml:1]: Please add license header

File: tools/quota-monitoring-alerting/python/templates/inputs/quota_email_notification.yaml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+description: Primary contact method when Quota issues happen
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/python/templates/inputs/quota_email_notification.yaml:1]: Please add license header

**Review by boredabdel — COMMENTED** [review]
Hi, Thanks for the PR. I would please ask you next time to split it into smaller PR's as reviewing an 83 files PR is time consuming.

I have added few comments. Please have a look and i will do an other round of review later on

File: tools/quota-monitoring-alerting/python/bigquery_schemas/dashboard_view.sql (hunk 2 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/bigquery_schemas/dashboard_view.sql:14]: Done

File: tools/quota-monitoring-alerting/python/config.yaml (hunk 3 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/config.yaml:13]: Done

File: tools/quota-monitoring-alerting/python/main.py (hunk 6 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/main.py:106]: Not at this point. But I see what you are saying. Can I postpone this correction to the next PR?

File: tools/quota-monitoring-alerting/python/main.py (hunk 7 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/main.py:112]: Not at this point. But I see what you are saying. Can I postpone this correction to the next PR?

File: tools/quota-monitoring-alerting/python/requirements.txt (hunk 5 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/requirements.txt:12]: Done

File: tools/quota-monitoring-alerting/python/templates/inputs/quota_email_notification.yaml (hunk 9 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/templates/inputs/quota_email_notification.yaml:13]: Done

File: tools/quota-monitoring-alerting/python/templates/inputs/quota_exceeded_threshold_report_policy.yaml (hunk 8 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/templates/inputs/quota_exceeded_threshold_report_policy.yaml:13]: Done

File: tools/quota-monitoring-alerting/python/Dockerfile (hunk 1 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/Dockerfile:41]: I am looking into this. Will update soon.

**Comment by r-kyatham** [comment]
> Hi, Thanks for the PR. I would please ask you next time to split it into smaller PR's as reviewing an 83 files PR is time consuming.
> 
> I have added few comments. Please have a look and i will do an other round of review later on

Hi, Thanks for your time on this PR and my apologies for the large PR.  Point noted.

PS: I know this is not a good answer but you can ignore all the files under 'java' directory.

File: tools/quota-monitoring-alerting/python/Dockerfile (hunk 1 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/Dockerfile]: Done.

File: tools/quota-monitoring-alerting/python/config.yaml (hunk 4 above)
Comment (r-kyatham) [reply tools/quota-monitoring-alerting/python/config.yaml]: Done

File: tools/quota-monitoring-alerting/python/main.py (hunk 7 above)
Comment (boredabdel) [reply tools/quota-monitoring-alerting/python/main.py:112]: Sure.

**Review by boredabdel — APPROVED** [review]
LGTM

**Pull request merged**